openmc/deplete/openmc_operator.py

```diff
@@ -0,0 +1,567 @@
+"""OpenMC transport operator
+
+This module implements functions used by both OpenMC transport operators as well as pure depletion operators.
+
+"""
+
+from abc import abstractmethod
+from collections import OrderedDict
+
+import numpy as np
+
+import openmc
+from openmc.mpi import comm
+from .abc import TransportOperator, OperatorResult
+from .atom_number import AtomNumber
+from .reaction_rates import ReactionRates
+
+__all__ = ["OpenMCOperator", "OperatorResult"]
+
+
+def _distribute(items):
+    """Distribute items across MPI communicator
+
+    Parameters
+    ----------
+    items : list
+        List of items of distribute
+
+    Returns
+    -------
+    list
+        Items assigned to process that called
+
+    """
+    min_size, extra = divmod(len(items), comm.size)
+    j = 0
+    for i in range(comm.size):
+        chunk_size = min_size + int(i < extra)
+        if comm.rank == i:
+            return items[j:j + chunk_size]
+        j += chunk_size
+
+
+class OpenMCOperator(TransportOperator):
+    """Abstract class holding OpenMC-specific functions for running
+    depletion calculations.
+
+    Specific classes for running coupled transport-depletion calculations or
+    depletion-only calculations are implemented as subclasses of OpenMCOperator
+
+    Parameters
+    ----------
+    materials : openmc.Materials
+        List of all materials in the model
+    cross_sections : str or pandas.DataFrame
+        Path to continuous energy cross section library, or object containing
+        one-group cross-sections.
+    chain_file : str, optional
+        Path to the depletion chain XML file.  Defaults to the file
+        listed under ``depletion_chain`` in
+        :envvar:`OPENMC_CROSS_SECTIONS` environment variable.
+    prev_results : Results, optional
+        Results from a previous depletion calculation. If this argument is
+        specified, the depletion calculation will start from the latest state
+        in the previous results.
+    diff_burnable_mats : bool, optional
+        Whether to differentiate burnable materials with multiple instances.
+        Volumes are divided equally from the original material volume.
+        Default: False.
```

Comment:
Defaults are listed in the function signature so not really necessary to put them in the docstring
```suggestion
```